rtengine/dcrop.cc

```diff
@@ -1476,6 +1462,97 @@ void Crop::update(int todo)
         
         parent->ipf.softLight(labnCrop, params.softlight);
 
+        if (params.icm.workingTRC != "none") {
+            int GW = labnCrop->W;
+            int GH = labnCrop->H;
+            LabImage *provis = nullptr;
+            float pres = 0.01f * params.icm.preser;
+            if(pres > 0.f) {
+                provis = new LabImage(GW, GH);
+                provis->CopyFrom(labnCrop);
+            }
+
+            Imagefloat *tmpImage1 = nullptr;
+            tmpImage1 = new Imagefloat(GW, GH);
+
+            parent->ipf.lab2rgb(*labnCrop, *tmpImage1, params.icm.workingProfile);
+
+            const float gamtone = parent->params->icm.workingTRCGamma;
+            const float slotone = parent->params->icm.workingTRCSlope;
+            //printf("Dcrop gam=%f slo=%f\n",gamtone, slotone); 
+            int illum = 0;
+            if(params.icm.will == "def"){
+                illum = 0; 
+            } else if(params.icm.will == "D41"){
+                illum = 1; 
+            } else if(params.icm.will == "D50"){
+                illum = 2; 
+            } else if(params.icm.will == "D55"){
+                illum = 3; 
+            } else if(params.icm.will == "D60"){
+                illum = 4; 
+            } else if(params.icm.will == "D65"){
+                illum = 5; 
+            } else if(params.icm.will == "D80"){
+                illum = 6; 
+            } else if(params.icm.will == "D120"){
+                illum = 7; 
+            } else if(params.icm.will == "stda"){
+                illum = 8; 
+            } else if(params.icm.will == "2000"){
+                illum = 9; 
+            } else if(params.icm.will == "1500"){
+                illum = 10; 
+            }
+
+            int prim = 0;
+            if(params.icm.wprim == "def"){
+                prim = 0; 
+            } else if(params.icm.wprim == "srgb"){
+                prim = 1; 
+            } else if(params.icm.wprim == "adob"){
+                prim = 2; 
+            } else if(params.icm.wprim == "prop"){
+                prim = 3; 
+            } else if(params.icm.wprim == "rec"){
+                prim = 4; 
+            } else if(params.icm.wprim == "aces"){
+                prim = 5; 
+            } else if(params.icm.wprim == "wid"){
+                prim = 6; 
+            } else if(params.icm.wprim == "ac0"){
+                prim = 7; 
+            } else if(params.icm.wprim == "bru"){
+                prim = 8; 
+            } else if(params.icm.wprim == "bet"){
+                prim = 9; 
+            } else if(params.icm.wprim == "bst"){
+                prim = 10; 
+            } else if(params.icm.wprim == "cus"){
+                prim = 11; 
+            } else if(params.icm.wprim == "cusgr"){
+                prim = 12; 
+            }
```

Comment:
Switch statements would have been much nicer here, but this works of course. An enum somewhere that immediately maps meaningful variable names to integer values would probably be a good improvement in the future.

Reply:
It is volontary normal

Reply:
@Thanatomanic There's no switch on strings in C/C++, but having a `static const std::unordered_map<std::string, int>` would result in a nice and compact table with a relatively quick lookup, though speed doesn't matter here. :+1: for the enum.